### ISSUE TITLE
display warping issue

### ISSUE THREAD

**Issue by cezarbabin** [googlevr/gvr-unity-sdk#589]
When building the Unity App with : https://github.com/blitzagency/ios-unity5, I get a weird display warp issue. For reference see comparison below:

Unity App integrated in iOS:
![img_5953](https://cloud.githubusercontent.com/assets/7832880/26388389/f70e86e2-4021-11e7-9bb1-648b35b8f0f0.PNG)

Unity App direct build:
![img_5954](https://cloud.githubusercontent.com/assets/7832880/26388380/eb9db1b6-4021-11e7-970a-f763d6b3c784.PNG)

**Comment by thri11house** [comment]
I've seen the same thing in my builds.  Here's what I've found:

5.6.0f3 - iOS VR view looks fine.
Any of the 5.6.0 patches - iOS VR view is distorted.
5.6.1.f1 - iOS VR view is distorted.

This issue doesn't seem to be associated with GVR SDK 1.2,1.4, 1.5, 1.6, just Unity version.

Any one else seeing the same thing?

**Comment by guerwan** [comment]
Same issue for me with 5.6.1f1. Updating the GVR SDK to 1.60 in the pod configuration file doesn't fix the issue.

**Comment by jgnoel86** [comment]
I am seeing this as well. Current app which loads in using None as the sdk then loads into Cardboard shows distortion on 5.6.1f1. But just opening it with 5.6.0p1 corrects this.

**Comment by MrSilverstone** [comment]
Same issue with 5.6.1. Working with 5.6.0p1

**Comment by xiesi88** [comment]
Same issue with 2017.1.0b6

**Comment by xiesi88** [comment]
@thri11house Same issue with 2017.1.0b6. iOS VR view is distorted.

**Comment by thri11house** [comment]
Let's summarize the list.  

Undistorted iOS VR Cameras:
5.6.0f3
5.6.0p1

Distorted iOS VR Cameras:
5.6.0p2
5.6.0p3
5.6.0p4
5.6.1f1
2017.1.0b6

**Comment by hedbergj** [comment]
Yep, known and reported: 

https://issuetracker.unity3d.com/issues/ios-cardboard-left-and-right-camera-view-is-offset-too-far-away-from-one-another-in-cardboard-mode-on-iphone-5-slash-ipod-6th-gen

(it says it's fixed, but it's not yet)

https://forum.unity3d.com/threads/5-6-0p3-gvr-1-40-ios10-3-1-iphone5.468716/

The consensus seems to be that it's a unity issue, not a gvr-sdk.

**Comment by MrSilverstone** [comment]
Actually I think that this is not the issue that the op is experiencing.
I load Unity from a native app in a similar way as @cezarbabin. Even on Unity 5.6.0 there is an issue with the rendering. This is how I solved the problem : 

In Build Settings I added None and Carboard (in this order) as Virtual Reality SDKs.
Then I added this function in a script of a GameObject in my scene :

 ```
public void StartVR(string str)
    {
        Debug.Log("Loading cardboard settings...");
        StartCoroutine(LoadDevice("cardboard"));
    }

    IEnumerator LoadDevice(string newDevice)
    {
        VRSettings.LoadDeviceByName(newDevice);
        yield return null;
        VRSettings.enabled = true;
     }
```

Then in my ViewController (I'm using objective-c) I send a message to the app :

```
NSLog(@"Launching unity");

UnitySendMessage("SceneManager", "StartVR", "");
[(AppDelegate*)[UIApplication sharedApplication].delegate showUnityWindow];

```

**Comment by guneyozsan** [comment]
I saw this in 2017.1.0b8, is it related?
VR: Fix for iOS Cardboard Left and right camera view is offset too far away from one another due to wrong viewport settings (being backported to 5.6) (912866)

**Comment by thri11house** [comment]
Thank you, guneyozsan.  This is good news.  

Does anyone want to guess when it will be backported to 5.6?

**Comment by nathanmartz** [comment]
This should be fixed in 5.6.1p2

**Comment by thri11house** [comment]
Yes.  I can confirm that 5.6.1p2 resolves the distortion.  Any reason why 912866 wasn't listed as a fix?

thanks all.
